src/Init/Data/Int/Bitwise/Lemmas.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2023 Siddharth Bhat. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Siddharth Bhat, Jeremy Avigad
+-/
+prelude
+import Init.Data.Nat.Bitwise.Lemmas
+import Init.Data.Int.Bitwise
+
+namespace Int
+
+theorem shiftRight_eq (n : Int) (s : Nat) : n >>> s = Int.shiftRight n s := rfl
+@[simp]
+theorem shiftRight_ofNat (n s : Nat) : (n : Int) >>> s = Int.ofNat (n >>> s) := rfl
+theorem natCast_shiftRight (n s : Nat) : ((↑n) : Int) >>> s = n >>> s := rfl
+
+@[simp]
+theorem shiftRight_negSucc (m n : Nat) :
+    -[m+1] >>> n = -[m >>>n +1] := rfl
+
+/-- Equation theorem for 'Int.sub' when the arguments are `.ofNat` and
+  the result is known to be negative. -/
+private theorem toNat_sub_toNat_eq_negSucc_ofLt {n m : Nat} (hlt : n < m) :
+  (n : Int) - (m : Int) = (Int.negSucc (m - 1 - n)) := by
+  rw [Int.negSucc_eq] -- TODO: consider adding this to omega cleanup set.
+  omega
+
+/-- shiftRight of a subtraction is evaluated when
+  the result of the subtraction is negative. -/
+theorem toNat_sub_toNat_shiftRight_eq_ofLt {m n i: Nat} (h : m < n) :
+  ((m : Int) - (n : Int)) >>> i = -[((n - 1 - m) >>> i) +1] := by
+    rw [toNat_sub_toNat_eq_negSucc_ofLt (by omega)]
+    rw [shiftRight_negSucc]
+
+theorem shiftRight_shiftRight (i : Int) (m n : Nat) :
```

Comment:
This equality should be flipped and renamed to `shiftRight_add` for consistency with `Nat`.

Reply:
This breaks consistency with [`Nat.shiftLeft_shiftLeft`](https://github.com/leanprover/lean4/blob/873ef2d894af80d8fc672e35f7e28bae314a1f6f/src/Init/Data/Nat/Lemmas.lean#L788-L790), however.

Reply:
Sigh. :-) I think the flipped version is slightly more likely to be useful? Since we're going to be inconsistent either way, my slight preference is
* flip this one
* express my everlasting gratitude when you make a separate PR renaming Nat.shiftLeft_shiftLeft (with a deprecation warning).

But it's also fine to just leave as is.

Reply:
Renamed and flipped equality to `Int.shiftRight_add`, will address the rest in a follow up commit.